### PR TITLE
Updated forms to have better input validation

### PR DESCRIPTION
**Pull request by KavsHub**
**EDIT** I've committed a lot of small UAT and issues into this PR (most were just 1 line changes) so this PR contains a lot of content changes.
This addresses the wishes of  #113, #118, #126, #130, #131 and a bug fix in subplan creation that I found along the way.  (#149 #101)
I've added a lot of strict validation that follows the standards from Programs and Courses.
Firstly all codes are now saved as uppercase, even if the user enters in lowercase or mixed.
All units must be divisible by 6.

**SUBPLANS**
Subplan now checks for the plan type in the course code
Capture
Subplan codes should have a '-' to split its plan type
Capture2
Subplans names shouldn't have the plan type as the code is made for that #131 
Capture3
I actually found this bug and fixed it, if nothing is selected we get a keyerror. I fixed this by catching the error and notifying the user to fill all fields
Capture4

**COURSES**
We now have very strict course codes which follow the current P&C syntax
If not 4 letters at the start
Capture5
if not 4 numbers after
Capture6
if code is too long or short
Capture7
I added this because codes are not all length 8, some codes have A, B C on them so I added an optional letter for it
Capture9

Please tell me what you think. I think this is good for ensuring accurate data and I've followed all the Programs and courses standards.

Also, let me know if theres any more strict checks you want me to do.

### CODE REVIEW

File: cassdegrees/ui/forms.py [hunk 1]
```diff
@@ -151,9 +185,17 @@ class Meta:
 
     def clean_code(self):
         data = self.cleaned_data['code']
-        if len(data) < 4:
-            raise forms.ValidationError("This should be at least 4 characters!")
-        return data
+        print(len(data))
```
Comment (j-selby) [inline cassdegrees/ui/forms.py]: Unneeded print statement

**Review by j-selby — CHANGES_REQUESTED** [review]
Looks good.

File: cassdegrees/ui/forms.py (hunk 1 above)
Comment (KavsHub) [reply cassdegrees/ui/forms.py]: Thanks haha.

**Review by JackSekoranja — APPROVED** [review]
Looks good. I should mention that CASS currently offers a Major with the word 'Major' in the title ([Animation and Video Major](https://programsandcourses.anu.edu.au/2019/major/ANVI-MAJ)), so we wouldn't be able to add that to the system currently. However, I understand this was a specific UAT request, so it may be a good idea to bring this up next client meeting (and admittedly this seems to be the only Subplan with the Subplan type in the name).

**Comment by KavsHub** [comment]
> Looks good. I should mention that CASS currently offers a Major with the word 'Major' in the title ([Animation and Video Major](https://programsandcourses.anu.edu.au/2019/major/ANVI-MAJ)), so we wouldn't be able to add that to the system currently. However, I understand this was a specific UAT request, so it may be a good idea to bring this up next client meeting (and admittedly this seems to be the only Subplan with the Subplan type in the name).

I feel like that was a CASS mistake haha

**Comment by KavsHub** [comment]
This PR addresses 6 issues now since I've combined a lot of small UAT issues and other issues in the backlog which were mostly 1 line changes or small fixes. 
Going to merge this once I've tested it thoroughly

**Pull request merged**